velox/vector/FlatVector.h

```diff
@@ -488,6 +495,12 @@ class FlatVector final : public SimpleVector<T> {
   }
 
  private:
+  void ensureValues() {
+    if (rawValues_ == nullptr) {
+      mutableRawValues();
+    }
```

Comment:
Shall we keep checking `VELOX_DCHECK(!values_->isView())` if `rawValues_` is not null?

Reply:
mutableRawValues() ensures that values is writable.

Reply:
We only call `mutableRawValues` when values is null

Reply:
I see. Let me add VELOX_DCHECK(!values_->isView()) back.